sdk/core/azure-core/azure/core/paging/_paging_method_handler.py

```diff
@@ -0,0 +1,110 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import logging
+
+from ..exceptions import (
+    map_error,
+    HttpResponseError,
+    ClientAuthenticationError,
+    ResourceExistsError,
+    ResourceNotFoundError
+)
+from ..pipeline import PipelineResponse
+
+_LOGGER = logging.getLogger(__name__)
+
+class _PagingMethodHandlerBase(object):
+    def __init__(
+        self,
+        paging_method,
+        deserialize_output,
+        client,
+        initial_state,
+        **kwargs
+    ):
+        self._paging_method = paging_method
+        self._deserialize_output = deserialize_output
+        self._client = client
+        self._initial_state = initial_state
+        self._cls = kwargs.pop("_cls", None)
+        self._error_map = {
+            401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError
+        }
+        self._error_map.update(kwargs.pop('error_map', {}))
+        self._item_name = kwargs.pop("item_name", "value")
+        self._continuation_token_location = kwargs.pop("continuation_token_location", None)
+        self._operation_config = kwargs
+
+    @property
+    def _initial_request(self):
```

Comment:
Are subclasses supposed to override any of the private methods or access any private attributes? If so, they should be made public.